jormungandr/src/rest/mod.rs

```diff
@@ -33,14 +33,30 @@ pub async fn start_rest_server(config: Rest, explorer_enabled: bool, context: Co
     let api = warp::path!("api" / ..)
         .and(v0::filter(context.clone()).or(v1::filter(context.clone())))
         .with(warp::filters::trace::trace(|info| {
-            tracing::span!(
+            use http_zipkin::get_trace_context;
+            use tracing::field::Empty;
+            let span = tracing::span!(
                 tracing::Level::DEBUG,
                 "rest_api_request",
                 method = %info.method(),
-                path = %info.path(),
+                path = info.path(),
                 version = ?info.version(),
-                remote_addr = ?info.remote_addr(),
-            )
+                remote_addr = Empty,
+                trace_id = Empty,
+                span_id = Empty,
+                parent_span_id = Empty,
+            );
+            if let Some(remote_addr) = info.remote_addr() {
+                span.record("remote_addr", &remote_addr.to_string().as_str());
```

Comment:
Heck, this API is bad. `str` implements `Value`, but you can't pass the result of `.as_str()` here because the `record` method requires reference to an implicitly `Sized` value.

Reply:
Yes, annoying as hell.